resources/metrics/profiling_shell.sh

```diff
@@ -1,45 +1,68 @@
 #!/bin/bash
 
-# this variant of the Profiling shell wrapper
-# uses only the Bash keyword `time`, and the generally
+# Vanagon profiling shell wrapper
+# version: 1.0.0
+# author: Ryan McKern <mckern@puppet.com>
+
+# The Vanagon Profiling shell wrapper
+# uses only the Bash keyword `time` and the generally
 # standard command `tee` to get run time. This was
 # needed because the BSD/sysv variants of `date` only
 # offer second-resolution. No milliseconds, no nano-
-# seconds.
-
-# Create copies of stdout & stderr's file descriptors
-# so that the output of the subshell that `time` runs
-# can be captured and used.
-exec 3>&1
-exec 4>&2
+# seconds. The results of `time` will be written to a
+# csv file containing a single column (times for each
+# subshell run as part of a given Make target. Each line
+# is a subshell's runtime in whole seconds with 3 decimal 
+# places of precision), named after the Make target, e.g.
+#   facter-configure.csv, facter-install.csv, etc.
+#
+# Of note is that we are intentionally avoiding using
+# `set -o pipefail` or `set -e` as these can cause
+# numerous false-positive failures depending on the
+# level of care and complexity in a given Make recipe.
 
 # Write some log files, that can be post-processed
-# relatively easily.
-__stdout="${VANAGON_TARGET}.stdout.log"
-__stderr="${VANAGON_TARGET}.stderr.log"
+# relatively easily. While you're at it, store runtimes in batches.
+if [[ -n ${VANAGON_TARGET} ]]; then
+  __batchfile="runtimes/${VANAGON_TARGET}.csv"
+  __stdout="logs/${VANAGON_TARGET}.stdout.log"
+  __stderr="logs/${VANAGON_TARGET}.stderr.log"
+  mkdir "logs" "runtimes"
+else
+  # Alternatively, if there's no defined Target, just
+  # dump that stuff in the bitbucket
+  __batchfile="/dev/null"
```

Comment:
There was a previous comment, might have been lost - how does /dev/null resolve on windows? Do we depend in cygwin on the build environment?